File: Source/AlphaTab/Rendering/Glyphs/DrumSticksGlyph.cs
namespace AlphaTab.Rendering.Glyphs
{
    class DrumSticksGlyph : MusicFontGlyph
    {
        private readonly bool _isGrace;

        public DrumSticksGlyph(float x, float y, bool isGrace)
            : base(x, y, isGrace ? NoteHeadGlyph.GraceScale : 1, MusicFontSymbol.NoteSideStick)
        {
            _isGrace = isGrace;
        }

        public override void DoLayout()
        {
            Width = 9 * (_isGrace ? NoteHeadGlyph.GraceScale : 1) * Scale;
        }
    }
}

File: src/rendering/glyphs/DrumSticksGlyph.ts
import { MusicFontGlyph } from '@src/rendering/glyphs/MusicFontGlyph';
import { MusicFontSymbol } from '@src/rendering/glyphs/MusicFontSymbol';
import { NoteHeadGlyph } from '@src/rendering/glyphs/NoteHeadGlyph';

export class DrumSticksGlyph extends MusicFontGlyph {
    private _isGrace: boolean;

    public constructor(x: number, y: number, isGrace: boolean) {
        super(x, y, isGrace ? NoteHeadGlyph.GraceScale : 1, MusicFontSymbol.NoteSideStick);
        this._isGrace = isGrace;
    }

    public doLayout(): void {
        this.width = 9 * (this._isGrace ? NoteHeadGlyph.GraceScale : 1) * this.scale;
    }
}
